File: LeetCode/Easy/0100-same-tree/0100-same-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        Queue<Integer> OrderP = new ArrayDeque<>();
        Queue<Integer> OrderQ = new ArrayDeque<>();
        trevasalTree(p, OrderP);
        trevasalTree(q, OrderQ);

        if (OrderP.size() != OrderQ.size()) return false;

        while (!OrderP.isEmpty() && !OrderQ.isEmpty()){ 
            if (!OrderP.poll().equals(OrderQ.poll())){
                return false;
            }
        }

        return true;
    }
    public void trevasalTree(TreeNode t, Queue<Integer> queue){
        // System.out.print(t.val+ " ");
        if (t != null){
            queue.add(t.val);
        } else {
            return;
        }

        if (t.left != null){
            trevasalTree(t.left, queue);
        } else {
            queue.add(100000);
        }
        
        if (t.right != null){
            trevasalTree(t.right, queue);
        } else {
            queue.add(100000);
        }
    }
}

File: LeetCode/Easy/0100-same-tree/0100-same-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isSameTree(p *TreeNode, q *TreeNode) bool {
    pVals := make([]int, 0)
    qVals := make([]int, 0)

    bfs(p, &pVals)
    bfs(q, &qVals)

    pLen := len(pVals)
    qLen := len(qVals)

    fmt.Println(pVals)
    fmt.Println(qVals)
    if (pLen != qLen) {
        return false
    }

    for i := 0; i < pLen; i++ {
        if (pVals[i] != qVals[i]) {
            return false
        }
    }

    return true
}

func bfs(p *TreeNode, pVals *[]int) {
    st := make([]*TreeNode, 0)
    st = append(st, p)
    if (p!= nil){
        *pVals = append(*pVals, p.Val)
    } else {
        *pVals = append(*pVals, 100000)
    }
    

    for (len(st) != 0) {
        cur := st[len(st)-1]
        st = st[:len(st)-1]

        if (cur == nil) {
            continue
        }

        if (cur.Left != nil) {
            st = append(st, cur.Left)
            *pVals = append(*pVals, cur.Left.Val)
        } else {
            st = append(st, cur.Left)
            *pVals = append(*pVals, 100000)
        }

        if (cur.Right != nil) {
            st = append(st, cur.Right)
            *pVals = append(*pVals, cur.Right.Val)
        } else {
            st = append(st, cur.Right)
            *pVals = append(*pVals, 100000)
        }
    }
}